oak/server/rust/oak_runtime/src/channel.rs

```diff
@@ -230,6 +251,18 @@ impl ChannelReader {
             }
         }
     }
+
+    /// Insert the given `thread` reference into `thread_id` slot of the HashMap of waiting
+    /// channels attached to an underlying channel. This allows the channel to wake up any waiting
+    /// channels by calling `thread::unpark` on all the threads it knows about.
+    pub fn add_waiter(
```

Comment:
Could maybe make this something like `add_waiter_current_thread(&self)` and do `let thread_ref = Arc::downgrade(Arc::new(platform::thread::current()));` inside the function?  

Dunno if that would make a `wait_on_channels` less efficient though &ndash; is `platform::thread::current()` expensive?

Reply:
I don't think this would work- `thread::current()` just returns a `Thread` which we wrap in `Arc`. Scoping it here would result in it being dropped immediately at the end of this function. Creating it inside the loop in `wait_on_channels` allows us to scope it to a single loop iteration, giving us tight control of when the references in various `Channel`s go stale (preventing too many spurious `unpark`s).

Reply:
Ah yes, I forgot about the lifetime side of things.